waimak-core/src/main/scala/com/coxautodata/waimak/dataflow/SequentialDataFlowExecutor.scala

```diff
@@ -37,7 +38,10 @@ class SequentialDataFlowExecutor[T, C](override val flowReporter: FlowReporter[T
       logInfo(s"Submitting action ${action.logLabel}")
       //TODO: left for compatibility, need to change the data flow entities to know about optional
       flowReporter.reportActionStarted(action, dataFlow.flowContext)
-      val actionOutputs: Seq[Option[T]] = action.performAction(inputEntities, dataFlow.flowContext)
+      val actionOutputs: Seq[Option[T]] = action.performAction(inputEntities, dataFlow.flowContext) match {
```

Comment:
@alexeipab mentioned this should be done on the trait for a unified way of unpacking and handling the errors. Will make the change and update.